src/ssl/ProxyCerts.h

```diff
@@ -11,22 +11,49 @@
 
 #if USE_OPENSSL
 #include "acl/forward.h"
+#include "acl/Gadgets.h"
+#include "ssl/gadgets.h"
 
 class sslproxy_cert_sign
 {
 public:
-    int alg;
-    ACLList *aclList;
-    sslproxy_cert_sign *next;
+    sslproxy_cert_sign() = default;
+    sslproxy_cert_sign(sslproxy_cert_sign &&) = delete; // prohibit all copy/move
+    ~sslproxy_cert_sign() {
+        if (aclList)
+            aclDestroyAclList(&aclList);
+        while (const auto first = next) {
+            next = first->next;
+            delete first;
+        }
+    }
+
+public:
+    Ssl::CertSignAlgorithm alg = Ssl::algSignEnd;
+    ACLList *aclList = nullptr;
+    sslproxy_cert_sign *next = nullptr;
 };
 
 class sslproxy_cert_adapt
 {
 public:
-    int alg;
-    char *param;
-    ACLList *aclList;
-    sslproxy_cert_adapt *next;
+    sslproxy_cert_adapt() = default;
+    sslproxy_cert_adapt(sslproxy_cert_adapt &&) = delete; // prohibit all copy/move
+    ~sslproxy_cert_adapt() {
+        xfree(param);
+        if (aclList)
+            aclDestroyAclList(&aclList);
+        while (const auto first = next) {
+            next = first->next;
+            delete first;
+        }
```

Comment:
I do not insist on this change because I am worried about creating a precedent that will trigger incorrect change requests in future reviews, but this destructor code will be safer if we delete `next` objects _before_ destroying parts of our object (and, hence, entering a dangerous inconsistent state):

```suggestion
        while (const auto first = next) {
            next = first->next;
            delete first;
        }
        if (aclList)
            aclDestroyAclList(&aclList);
        xfree(param);
```

The above order also matches the default C++ class member destruction order, which is usually unimportant, but is a nice bonus.

Same for sslproxy_cert_sign, of course.

Reply:
Sure. Done in 98298ed. Also added the missing nullptr assignment